backend/app/src/main/java/ch/zhaw/studyflow/controllers/CalendarController.java

```diff
@@ -0,0 +1,328 @@
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            request.getRequestBody()
+                    .flatMap(body -> body.tryRead(Calendar.class))
+                    .flatMap(calendar -> {
+                        calendarManager.create(calendar);
+                        return Optional.of(calendar);
+                    })
+                    .ifPresentOrElse(
+                            calendar -> {
+                                response.setResponseBody(JsonContent.writableOf(calendar))
+                                        .setStatusCode(HttpStatusCode.CREATED);
+                            },
+                            () -> {
+                                response.setStatusCode(HttpStatusCode.BAD_REQUEST);
+                            }
+                    );
+            return response;
+        });
+    }
+
+    /**
+     * Endpoint for creating a new appointment in a specific calendar.
+     *
+     * @param context the request context
+     * @return the HTTP response
+     */
+    @Route(path = "{calendarId}/appointments")
+    @Endpoint(method = HttpMethod.POST)
+    public HttpResponse createAppointment(RequestContext context) {
+        final HttpRequest request = context.getRequest();
+        HttpResponse response = context.getRequest().createResponse();
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            request.getRequestBody()
+                    .flatMap(body -> body.tryRead(Appointment.class))
+                    .flatMap(appointment -> {
+                        appointmentManager.create(appointment);
+                        return Optional.of(appointment);
+                    })
+                    .ifPresentOrElse(
+                            appointment -> {
+                                response.setResponseBody(JsonContent.writableOf(appointment))
+                                        .setStatusCode(HttpStatusCode.CREATED);
+                            },
+                            () -> {
+                                response.setStatusCode(HttpStatusCode.BAD_REQUEST);
+                            }
+                    );
+            return response;
+        });
+    }
+
+    /**
+     * Endpoint for retrieving a specific calendar.
+     *
+     * @param context the request context
+     * @return the HTTP response
+     */
+    @Route(path = ":id")
+    @Endpoint(method = HttpMethod.GET)
+    public HttpResponse getCalendar(RequestContext context) {
+        final HttpRequest request = context.getRequest();
+        HttpResponse response = context.getRequest().createResponse();
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            request.getRequestBody()
+                    .flatMap(body -> body.tryRead(Calendar.class))
+                    .flatMap(calendar -> {
+                        Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID).map(Long::valueOf);
```

Comment:
`Long.valueOf` throws a `NumberFormatException `if the provided value cannot be parsed. To handle this, we should either manage the exception appropriately or use an alternative parsing method that avoids exceptions. In either case, the response should return BAD_REQUEST instead of allowing the connection to fail.